### PR TITLE
Rename package to DependencyGraph in code

### PR DESCRIPTION
**Pull request by DivineDominion**
Snake Case `swift_dependency_graphs` is rather uncommon

### CODE REVIEW

File: Package.swift [hunk 1]
```diff
@@ -4,13 +4,13 @@
 import PackageDescription
 
 let package = Package(
-    name: "swift-dependency-graphs",
```
Comment (ikelax) [inline Package.swift:7]: For the name we should use kebab case similar to [swift-argument-parser](https://github.com/apple/swift-argument-parser/blob/main/Package.swift) and [swift-syntax](https://github.com/swiftlang/swift-syntax/blob/main/Package.swift).

**Comment by ikelax** [comment]
Thank you for the contribution! I left a comment 🙂

File: Package.swift (hunk 1 above)
Comment (DivineDominion) [reply Package.swift:7]: Makes sense. Feel free to edit the PR

Comment (ikelax) [reply Package.swift]: I applied the change.

**Review by ikelax — APPROVED** [review]

**Pull request merged**